File: frontend/src/mappers/AuctionMapper.js
import AuctionDto from "../dto/auction/auctionDto";

export default class AuctionMapper {

    static ToAuctionDto(data) {
        let auctionDto = new AuctionDto();
        auctionDto.id = data.id;
        auctionDto.auctionStartTime = new Date(data.auctionStartTime);
        auctionDto.auctionEndTime = new Date(data.auctionEndTime);
        auctionDto.winnerId = data.winnerId;
        auctionDto.highestBidAmount = data.highestBidAmount;
        auctionDto.isActive = data.isActive;
        auctionDto.carId = data.carId;
        auctionDto.car = data.car ? data.car : null;
        auctionDto.bids = data.bids ? data.bids : null;
        return auctionDto;
    }
}

File: backend/Mappers/AuctionMapper.cs
using DreamBid.Dtos.Auction;
using DreamBid.Models;

namespace DreamBid.Mappers
{
    public static class AuctionMapper
    {
        public static Auction ToAuctionFromAddAuctionDto(this AddAuctionDto addAuctionDto)
        {
            return new Auction
            {
                auctionStartTime = (DateTime)addAuctionDto.AuctionStartTime,
                auctionEndTime = (DateTime)addAuctionDto.AuctionEndTime,
                IsActive = false,
            };
        }

        public static AuctionDto ToAuctionDto(this Auction auction)
        {
            return new AuctionDto
            {
                auctionEndTime = auction.auctionEndTime,
                auctionStartTime = auction.auctionStartTime,
                carId = auction.CarId,
                highestBidAmount = auction.highestBidAmount,
                id = auction.Id,
                isActive = auction.IsActive,
                winnerId = auction.WinnderId,
                Bids = auction.Bids.Select(b => b.ToBidDto()).ToList(),
                Car = auction.Car != null ? auction.Car.ToCarDto() : null
            };
        }
    }
}